### ISSUE TITLE
Media add/edit/delete broke somewhere between Drupal 9.4.0-dev and 9.4.2

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#446]
Somewhere after Drupal 9.4.0-dev and Drupal 9.4.2, the REST endpoint for reading/editing/deleting media changed:

Drupal 9.4.0-dev:

![media_rest_endpoint_drupal_940dev](https://user-images.githubusercontent.com/403918/180454108-1ffc721c-88ce-4fce-8251-32c5db768afc.png)

9.4.2 (GET, PATCH, DELETE now has "edit" at the end):

![media_rest_endpoint_drupal_942](https://user-images.githubusercontent.com/403918/180454150-ea70c97b-1d1e-4984-a160-7ea69ce7df92.png)

I cannot find any mention of this in the release notes. FTW semantic versioning.

Since Workbench uses this endpoint, logic will need to be added to account for the change.

**Comment by mjordan** [comment]
The d.o. issue is https://www.drupal.org/project/drupal/issues/3017935.

**Comment by mjordan** [comment]
Resolved with 029cb6d51e43a7226f12597fe379076e537a91c4.

Docs on configuring Drupal's media endpoints are at https://mjordan.github.io/islandora_workbench_docs/installation/#configuring-drupals-media-urls.

**Comment by dara2** [comment]
Hi @mjordan - in Slack you'd noted:

> Just merged this fix, after testing on our new IR (deleted media of 460 nodes) with no problems. Steps to update:
> 1. pull in updates to the main branch
> 1. visit /admin/config/media/media-settings and make sure "Standalone media URL" is unchecked (this is the Drupal default, so you should not need to change it). If you need to have this checked for some reason, please leave a comment on https://github.com/mjordan/islandora_workbench/issues/446 and we'll be able to support both types of media URIs.
> if you do need to change that setting, clear your Drupal cache.

We actually do need that box to be checked. If we un-check it, we can't reach the media derivatives (can only get to the "edit" view of them). How much work is it to support that?

**Comment by mjordan** [comment]
I don't know. I will look into it later today.

**Comment by mjordan** [comment]
@dara2 can you check out the issue-446 branch to test? You will need to add this to your config file: `standalone_media_url: true`. You will also need to have the "Standalone media URL" option in /admin/config/media/media-settings checked, and clear your Drupal cache.

**Comment by dara2** [comment]
Thanks, Mark! I'll test it today!

**Comment by mjordan** [comment]
Small world, I hit that today myself and am looking into it. Thanks for letting me know it's not just me..... stay tuned.

**Comment by mjordan** [comment]
I'll add, it's not related to the changes in the 446 branch, I'm running Workbench in the main branch.

**Comment by mjordan** [comment]
@margaret-bd I've fixed the bug I introduced a couple day ago that caused the error you encountered. Can you pull in changes to the issue-446 branch and retest?

**Comment by dara2** [comment]
Hi @mjordan - Thank you for the update! I tested yesterday and was unable to ingest PDFs. Each time it hit a PDF, Workbench got hung up for 15-20 minutes, and in the Drupal log I could see a PHP warning about file hash, hundreds of warnings about trying to access a media file that was no longer there (and wasn't named anything to do with what I was ingesting), and then the same 3 FITS files being created over and over. I suspect the issue has something to do with our setup, rather than Workbench, but we're investigating. I'll update you when I have more info.

Here's the PHP warning:
```
Type	php
Date	Thursday, August 11, 2022 - 15:16
User	admin
Location	https://islandora-staging.usainteanne.ca/entity/media
Referrer	
Message	Warning: hash_file(fedora://2022-08/DAFH_PERRET, Annie 2 2019-10-28_1.pdf): failed to open stream: "Drupal\flysystem\FlysystemBridge::stream_open" call failed in Drupal\filehash\FileHash->hash() (line 249 of /var/www/drupal/web/modules/contrib/filehash/src/FileHash.php)
#0 /var/www/drupal/web/core/includes/bootstrap.inc(347): _drupal_error_handler_real()
#1 [internal function]: _drupal_error_handler()
#2 /var/www/drupal/web/modules/contrib/filehash/src/FileHash.php(249): hash_file()
#3 /var/www/drupal/web/modules/contrib/filehash/src/FileHash.php(219): Drupal\filehash\FileHash->hash()
#4 /var/www/drupal/web/modules/contrib/filehash/filehash.module(72): Drupal\filehash\FileHash->filePresave()
#5 [internal function]: filehash_file_presave()
#6 /var/www/drupal/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(426): call_user_func_array()
#7 /var/www/drupal/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(405): Drupal\Core\Extension\ModuleHandler->Drupal\Core\Extension\{closure}()
#8 /var/www/drupal/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(433): Drupal\Core\Extension\ModuleHandler->invokeAllWith()
#9 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php(249): Drupal\Core\Extension\ModuleHandler->invokeAll()
#10 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php(903): Drupal\Core\Entity\EntityStorageBase->invokeHook()
#11 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php(563): Drupal\Core\Entity\ContentEntityStorageBase->invokeHook()
#12 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php(756): Drupal\Core\Entity\EntityStorageBase->doPreSave()
#13 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php(517): Drupal\Core\Entity\ContentEntityStorageBase->doPreSave()
#14 /var/www/drupal/web/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php(802): Drupal\Core\Entity\EntityStorageBase->save()
#15 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityBase.php(339): Drupal\Core\Entity\Sql\SqlContentEntityStorage->save()
#16 /var/www/drupal/web/modules/contrib/filehash/src/FileHash.php(206): Drupal\Core\Entity\EntityBase->save()
#17 /var/www/drupal/web/modules/contrib/filehash/filehash.module(36): Drupal\filehash\FileHash->entityStorageLoad()
#18 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php(870): filehash_entity_storage_load()
#19 /var/www/drupal/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(405): Drupal\Core\Entity\ContentEntityStorageBase->Drupal\Core\Entity\{closure}()
#20 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentEntityStorageBase.php(871): Drupal\Core\Extension\ModuleHandler->invokeAllWith()
#21 /var/www/drupal/web/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php(394): Drupal\Core\Entity\ContentEntityStorageBase->invokeStorageLoadHook()
#22 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php(346): Drupal\Core\Entity\Sql\SqlContentEntityStorage->doLoadMultiple()
#23 /var/www/drupal/web/core/lib/Drupal/Core/Entity/EntityStorageBase.php(296): Drupal\Core\Entity\EntityStorageBase->loadMultiple()
#24 /var/www/drupal/web/core/lib/Drupal/Core/Entity/Plugin/DataType/EntityReference.php(72): Drupal\Core\Entity\EntityStorageBase->load()
#25 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/DataReferenceBase.php(37): Drupal\Core\Entity\Plugin\DataType\EntityReference->getTarget()
#26 /var/www/drupal/web/core/lib/Drupal/Core/Field/FieldItemBase.php(140): Drupal\Core\TypedData\DataReferenceBase->getValue()
#27 /var/www/drupal/web/core/lib/Drupal/Core/Entity/Plugin/Validation/Constraint/ReferenceAccessConstraintValidator.php(27): Drupal\Core\Field\FieldItemBase->__get()
#28 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveContextualValidator.php(196): Drupal\Core\Entity\Plugin\Validation\Constraint\ReferenceAccessConstraintValidator->validate()
#29 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveContextualValidator.php(148): Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateConstraints()
#30 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveContextualValidator.php(158): Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateNode()
#31 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveContextualValidator.php(158): Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateNode()
#32 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveContextualValidator.php(100): Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateNode()
#33 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/Validation/RecursiveValidator.php(93): Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validate()
#34 /var/www/drupal/web/core/lib/Drupal/Core/TypedData/TypedData.php(132): Drupal\Core\TypedData\Validation\RecursiveValidator->validate()
#35 /var/www/drupal/web/core/lib/Drupal/Core/Entity/ContentEntityBase.php(489): Drupal\Core\TypedData\TypedData->validate()
#36 /var/www/drupal/web/core/modules/media/src/Entity/Media.php(415): Drupal\Core\Entity\ContentEntityBase->validate()
#37 /var/www/drupal/web/core/modules/rest/src/Plugin/rest/resource/EntityResourceValidationTrait.php(37): Drupal\media\Entity\Media->validate()
#38 /var/www/drupal/web/core/modules/rest/src/Plugin/rest/resource/EntityResource.php(184): Drupal\rest\Plugin\rest\resource\EntityResource->validate()
#39 [internal function]: Drupal\rest\Plugin\rest\resource\EntityResource->post()
#40 /var/www/drupal/web/core/modules/rest/src/RequestHandler.php(219): call_user_func_array()
#41 /var/www/drupal/web/core/modules/rest/src/RequestHandler.php(68): Drupal\rest\RequestHandler->delegateToRestResourcePlugin()
#42 [internal function]: Drupal\rest\RequestHandler->handle()
#43 /var/www/drupal/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(123): call_user_func_array()
#44 /var/www/drupal/web/core/lib/Drupal/Core/Render/Renderer.php(564): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#45 /var/www/drupal/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(124): Drupal\Core\Render\Renderer->executeInRenderContext()
#46 /var/www/drupal/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(97): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext()
#47 /var/www/drupal/vendor/symfony/http-kernel/HttpKernel.php(158): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#48 /var/www/drupal/vendor/symfony/http-kernel/HttpKernel.php(80): Symfony\Component\HttpKernel\HttpKernel->handleRaw()
#49 /var/www/drupal/web/core/lib/Drupal/Core/StackMiddleware/Session.php(58): Symfony\Component\HttpKernel\HttpKernel->handle()
#50 /var/www/drupal/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(48): Drupal\Core\StackMiddleware\Session->handle()
#51 /var/www/drupal/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(106): Drupal\Core\StackMiddleware\KernelPreHandle->handle()
#52 /var/www/drupal/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(85): Drupal\page_cache\StackMiddleware\PageCache->pass()
#53 /var/www/drupal/web/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(48): Drupal\page_cache\StackMiddleware\PageCache->handle()
#54 /var/www/drupal/web/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(51): Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle()
#55 /var/www/drupal/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Drupal\Core\StackMiddleware\NegotiationMiddleware->handle()
#56 /var/www/drupal/web/core/lib/Drupal/Core/DrupalKernel.php(709): Stack\StackedHttpKernel->handle()
#57 /var/www/drupal/web/core/assets/scaffold/files/index.php(19): Drupal\Core\DrupalKernel->handle()
#58 {main}
.
Severity	Warning
Hostname	47.51.79.50
Operations	
```

**Comment by dara2** [comment]
I also got a file error:
```
Type file
Date Thursday, August 11, 2022 - 15:16
User admin
Location https://islandora-staging.usainteanne.ca/file/upload/media/document/field_media_document?_format=json
Referrer
Message The file permissions could not be set on fedora://2022-08/DAFH_PERRET, Annie 2 2019-10-28_1.pdf.
Severity Error
Hostname 47.51.79.50
Operations
```

**Comment by dara2** [comment]
I didn't, but I just retested on another site that is also Drupal 9.4.3 and has the issue-446 branch of Workbench and it was ALL FINE. :) 
So I think something has gone awry on the other site, but not Workbench-related.

**Comment by mjordan** [comment]
Hard to say. If you can try to find out what the differences might be between the two sites, that might help us track down a solution. I don't think the specific Drupal version matters too much as long as it is at least 9.x, it's very likely a configuration difference.

**Comment by dara2** [comment]
Yeah, we're looking into it - I think you can consider this issue resolved, for Workbench purposes (but let me know if you disagree, and there is anything to look into involving file hash). The other site is multilingual, so there are a lot of config differences. It's also always had a problem related to FITS files "over-creating" themselves. We will likely end up updating containers and seeing if that resolves it. 
Thank you for your quick response!

**Comment by mjordan** [comment]
Thanks. I'll reclose this issue, but I'd be interested in hearing how you resolved the issue (no need to post here if you'd prefer not to) in case it will be useful to others who might encounter it.

**Comment by mjordan** [comment]
Merged into main with b82153320df5ba0966840e92a91d513be1c02d14.
